extra_data/read_machinery.py

```diff
@@ -107,9 +107,11 @@ def _tid_to_slice_ix(tid, train_ids, stop=False):
 
 
 def select_train_ids(train_ids, sel):
+    if isinstance(sel, by_id) and isinstance(sel.value, int):
```

Comment:
We should also allow numpy integers, because getting a trainID as a uint64 is a fairly likely possibility.

There's an [`operator.index()`](https://docs.python.org/3/library/operator.html#operator.index) function which tries to convert things to a Python integer for indexing, giving `TypeError` if it's not suitable.

Reply:
Another more manual option would be `numbers.Integral`:

```python3
In [5]: isinstance(5, numbers.Integral)
Out[5]: True

In [6]: isinstance(np.int64(5), numbers.Integral)
Out[6]: True
```

Reply:
Although it's not quite the same for 0D arrays:

```
In [18]: isinstance(np.array(4), numbers.Integral)
Out[18]: False

In [21]: operator.index(np.array(4))
Out[21]: 4
```

Reply:
fixed in 29a1528f290b5684fef78e285400e6fa340bbff4